src/App.tsx

```diff
@@ -1,25 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const TOTAL_ITEMS = 42;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+
+    handlePageChange(1);
+  };
+
+  const startItems = perPage * currentPage - (perPage - 1);
+  const endItems =
+    perPage * currentPage < TOTAL_ITEMS ? perPage * currentPage : TOTAL_ITEMS;
```

Comment:
The calculation of `endItems` is correct, but it should be noted that this variable is used incorrectly in the `items.map` function below. The logic should ensure that only items within the current page range are displayed.